src/runtime/types.h

```diff
@@ -371,14 +375,31 @@ class BoxedBool : public BoxedInt {
 
 class BoxedString : public Box {
 public:
-    // const std::basic_string<char, std::char_traits<char>, StlCompatAllocator<char> > s;
-    std::string s;
+    llvm::StringRef s;
+
+    char* storage() { return (char*)this + cls->tp_basicsize; }
```

Comment:
Hmm so some places call storage(), and some look at `->s.data()`.  I guess if we're going to store the pointer we should prefer to use it?  Maybe we could do something like have a private function that calculates the storage offset, and a public one that just returns s.data?